backend/src/src-cliservice/src/main/java/com/siemens/sw360/cliservice/parsers/CopyrightLicenseInfoParser.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Siemens AG, 2016. Part of the SW360 Portal Project.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package com.siemens.sw360.cliservice.parsers;
+
+import com.siemens.sw360.datahandler.couchdb.AttachmentConnector;
+import com.siemens.sw360.datahandler.thrift.attachments.AttachmentContent;
+import com.siemens.sw360.datahandler.thrift.cliservice.CopyrightLicenseInfo;
+
+/**
+ * @author: alex.borodin@evosoft.com
+ */
+public abstract class CopyrightLicenseInfoParser {
+    //    private static final Logger log = Logger.getLogger(CLIHandler.class);
+    public static final String FILETYPE_PARSING_IMPOSSIBLE = "NONE";
+    protected final AttachmentConnector attachmentConnector;
+
+    protected CopyrightLicenseInfoParser(AttachmentConnector attachmentConnector){
+        this.attachmentConnector = attachmentConnector;
+    }
+
+    public abstract boolean isApplicableTo(AttachmentContent attachmentContent);
+    public abstract CopyrightLicenseInfo getCLI(AttachmentContent attachmentContent);
```

Comment:
haha! naming discussion can be fun. :-)

Just a few thoughts on the purpose of this service:

* The ultimate solution for sw360 is to create a combined SPDX for the project or product and provide this with the customer. I am not sure you agree but at least that is my idea of it.
* Readme or disclosure document is just an artefact of the past, used for non-streamlined supply chains. The future is to have fully parse-able exchange formats. We use the readme now because we are so slow.
* Given the current service, why create another service for SPDX combined files then? You would use the same service and extend it, because half of the code is there anyways: look into the attachments of every involved release, parse the SPDX, create an aggregation based on the releases of a project.

Then with the naming:

* In the beginning there are service names, which were just substantives: Components, Projectsm Vendors, ...: nice and uniform.
* Actually, in the beginning we have had also the idea of a rest based API where the URL would result from such service, such as
GET http://localhost/components/mycomponent01 -> returns the JSON formatted dataset of mycomponents
GET http://localhost/components/ -> returns a listing of all component short names (IDs) etc
* PUT http:// ... can be done also nicely!
(* So in this world, how about bdpImport, why did not we call call projectsimport or import/projects, so you can mike a nice PUT request ...?)

Given these points, my thoughts on the naming for the current form:

* readme:  most people understand something else with readme than just license and copyright information. it is not the classic readme that introduces the software as first step. not so good.
* disclosure document: hm, maybe OK, but still too much to an artefact that seems to be organisation specific. Actually, I have never heard the term disclosure document before. hm.
* Component License Information: admitted a little bit bulky, we have hadused this term before. Maybe coincidental with another acronym. Frankly, I have seen so many projects with acronym overlap, I am not concerned much with the 427th occurrence of an acronym overlap. OK, maybe also not good.

I am voting for a name that reflects the future how about
* license information (easy compromise, just skip component)
* software data exchnage (too long)
* SPDX
* maybe if it is called licenseinformation it could be an idea to skip the word service as we have had also not -service suffix in the other services